---
title: "18 line enrichment (Aug 22 update)"
author: "Lifan Liang"
date: "2022-08-22"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

Like last time, I received DA peaks and cell type by time point peaks from two different procedures. One is from Signac. The other is from pseudobulk. 

## Peak count statistics
DA peaks from Signac (Lexi) are 2 time points (1hr and 6hr) by 3 cell types (GABA, NEFM pos glut, and NEFM neg glut). FDR cutoff was 0.05.
```{r, echo=F}
table <- read.table("../lexi/updatedcutoff_0819_fdr5/lexi_updatedcutoff_fdr5_0819.counts")
knitr::kable(table)
```

DA peaks from the pseudobulk approach
```{r, echo=F}
table <- read.table("../lexi/pseudobulk_0818_fdr5/lexi_pseudobulk_fdr5_0818.counts")
knitr::kable(table)
```


## LDSC joint enrichment analysis for Signac approach

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
#library(rtracklayer)

plot_enrichment = function(dat, title, xlab="Odds ratio", ylab="Annotations") {
  p1 <- ggplot(dat, aes(y=annotation)) + 
    geom_point(aes(x=Enrichment), colour="#800000") + #plot OR
    geom_segment(aes(x=lower,xend=upper,yend=annotation), alpha=0.8)
  p1 <- p1 + geom_vline(xintercept=0) + theme_bw() + labs(title=title, x=xlab, y=ylab)
  p1
}


plot_h2 = function(dat, title, xlab="Odds ratio", ylab="Annotations") {
  p1 <- ggplot(dat, aes(y=annotation)) + 
    geom_point(aes(x=Prop._h2), colour="#800000") + #plot OR
    geom_segment(aes(x=lower,xend=upper,yend=annotation), alpha=0.8)
  p1 <- p1 + geom_vline(xintercept=0) + theme_bw() + labs(title=title, x=xlab, y=ylab)
  p1
}
```

#### Enrichment analysis applied to various diseases
```{r, echo=F}
path <- "data/lexi_updatedcutoff_0819/"
res.dir <- list.files(path, pattern="results")
diseases <- sapply(strsplit(res.dir,"_"), function(x) {x[1]})
diseases[8] <- "insomnia"
diseases[9] <- "intelligence"
diseases[11] <- "SCZ"
aggr <- list()
for(i in seq_along(res.dir)) {
  enrich.ld <- read.table(paste0(path, res.dir[i]), sep="\t", header=T, row.names=1)
  temp <- enrich.ld[(nrow(enrich.ld)-8):nrow(enrich.ld),]
  annot <- c('NEFM_neg_glut_1v0hr',
 'NEFM_pos_glut_1v0hr',
 'GABA_1v0h',
 'Npglut_static',
 'NEFM_pos_glut_6v0hr',
 'GABA_6v0hr',
 'NEFM_neg_glut_6v0hr',
 'Nmglut_static',
 'GABA_static')
  rownames(temp) <- annot
  aggr[[i]] <- data.frame(disease=diseases[i], annot=annot, enrichment=temp$Enrichment, enrichment_pval=-log10(temp$Enrichment_p))
}
aggr <- do.call("rbind", aggr)
aggr$enrichment[aggr$enrichment<0] = 0

aggr$annot <- factor(aggr$annot, level=rev(annot[c(6,5,7,3,2,1,9,4,8)]))
aggr$disease <- factor(aggr$disease, level=diseases[c(11,2,5,7,8,9,10,1,6,4,12,3)])

#pdf("~/Documents/neuron_stim/Signac_various_diseases_enrichment.pdf", width = 9, height=5)
ggplot(data=aggr, aes(x=disease, y=annot, color=enrichment_pval, size=enrichment)) + 
  geom_point() +
  scale_color_gradient(low="black", high="red") +
  theme_bw() + 
  ylab("") + 
  xlab("") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(color="-log10(Pvalue)")
  ggtitle("S-LDSC enrichment analysis")
#dev.off()
```

#### Comparison of total heritability in schizophrenia (SCZ)
```{r, echo=F}
enrich.ld <- read.table("data/lexi_updatedcutoff_0819/PGC3_dyn_vs_stat_joint.results", sep="\t", header=T, row.names=1)
temp <- enrich.ld[(nrow(enrich.ld)-8):nrow(enrich.ld),]
### These labels and related levels should be used across all diseases
annot <- c('NEFM_neg_glut_1v0hr',
 'NEFM_pos_glut_1v0hr',
 'GABA_1v0h',
 'Npglut_static',
 'NEFM_pos_glut_6v0hr',
 'GABA_6v0hr',
 'NEFM_neg_glut_6v0hr',
 'Nmglut_static',
 'GABA_static')
lvls <- rev(c(6,3,9,5,2,4,7,1,8))
rownames(temp) <- annot
temp$annotation <- factor(rownames(temp), level=rownames(temp)[lvls])
temp$Prop._h2 <- 100 * temp$Prop._h2
temp$lower <- temp$Prop._h2 - 196 * temp$Prop._h2_std_error
temp$upper <- temp$Prop._h2 + 196 * temp$Prop._h2_std_error
plot_h2(temp, "", xlab = '% of SNP heritability explained', ylab = "Functional annotations")
```

#### Per SNP heritability (enrichment) from Signac (enrichment) in schizophrenia (SCZ)
```{r, echo=F}
enrich.ld <- read.table("data/lexi_updatedcutoff_0819/PGC3_dyn_vs_stat_joint.results", sep="\t", header=T, row.names=1)
temp <- enrich.ld[(nrow(enrich.ld)-8):nrow(enrich.ld),]
rownames(temp) <- annot
temp$annotation <- factor(rownames(temp), level=rownames(temp)[lvls])
temp$lower <- temp$Enrichment - 1.96 * temp$Enrichment_std_error
temp$upper <- temp$Enrichment + 1.96 * temp$Enrichment_std_error
plot_enrichment(temp, "", xlab = 'enrichment of SZ variants', ylab = "Functional annotations")
```




## LDSC joint enrichment analysis for various diseases GWAS from pseudoBulk
```{r, echo=F}
path <- "data/lexi_pseudobulk_0818/"
res.dir <- list.files(path, pattern="results")
diseases <- sapply(strsplit(res.dir,"_"), function(x) {x[1]})
diseases[8] <- "insomnia"
diseases[9] <- "intelligence"
diseases[11] <- "SCZ"
aggr <- list()
for(i in seq_along(res.dir)) {
  enrich.ld <- read.table(paste0(path, res.dir[i]), sep="\t", header=T, row.names=1)
  temp <- enrich.ld[(nrow(enrich.ld)-8):nrow(enrich.ld),]
  annot <- c(
 'nmglut_1v0hr',
 'GABA_1v0hr',
 'nmglut_6v0hr',
 'Npglut_static',
 'GABA_6v0hr',
 'Nmglut_static',
 'npglut_6v0hr',
 'GABA_static',
 'npglut_1v0hr')
  rownames(temp) <- annot
  aggr[[i]] <- data.frame(disease=diseases[i], annot=annot, enrichment=temp$Enrichment, enrichment_pval=-log10(temp$Enrichment_p))
}
aggr <- do.call("rbind", aggr)
aggr$enrichment[aggr$enrichment<0] = 0

aggr$annot <- factor(aggr$annot, level=annot[c(6,4,8,1,9,2,3,7,5)])
aggr$disease <- factor(aggr$disease, level=diseases[c(11,2,5,7,8,9,10,1,6,4,12,3)])

#pdf("~/Documents/neuron_stim/Signac_various_diseases_enrichment.pdf", width = 9, height=5)
ggplot(data=aggr, aes(x=disease, y=annot, color=enrichment_pval, size=enrichment)) + 
  geom_point() +
  scale_color_gradient(low="black", high="red") +
  theme_bw() + 
  ylab("") + 
  xlab("") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(color="-log10(Pvalue)")
  ggtitle("S-LDSC enrichment analysis")
#dev.off()
```

#### Comparison of total heritability in pseudo bulk (heritability) in schizophrenia (SCZ)
```{r, echo=F}
enrich.ld <- read.table("data/lexi_pseudobulk_0818/PGC3_dyn_vs_stat_joint.results", sep="\t", header=T, row.names=1)
temp <- enrich.ld[(nrow(enrich.ld)-8):nrow(enrich.ld),]
lvls <- c(6,1,3,4,9,7,8,2,5)
rownames(temp) <- annot
temp$annotation <- factor(rownames(temp), level=rownames(temp)[lvls])
temp$Prop._h2 <- 100 * temp$Prop._h2
temp$lower <- temp$Prop._h2 - 196 * temp$Prop._h2_std_error
temp$upper <- temp$Prop._h2 + 196 * temp$Prop._h2_std_error
plot_h2(temp, "", xlab = '% of SNP heritability explained', ylab = "Functional annotations")
```

#### Per SNP heritability (enrichment) from pseudo bulk  in schizophrenia (SCZ)
```{r, echo=F}
enrich.ld <- read.table("data/lexi_pseudobulk_0818/PGC3_dyn_vs_stat_joint.results", sep="\t", header=T, row.names=1)
temp <- enrich.ld[(nrow(enrich.ld)-8):nrow(enrich.ld),]
rownames(temp) <- annot
temp$annotation <- factor(rownames(temp), level=rownames(temp)[lvls])
temp$lower <- temp$Enrichment - 1.96 * temp$Enrichment_std_error
temp$upper <- temp$Enrichment + 1.96 * temp$Enrichment_std_error
plot_enrichment(temp, "", xlab = 'enrichment of SZ variants', ylab = "Functional annotations")
```





