---
title: "Enrichment analysis"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    number_sections: no
    toc: yes
    toc_float: true
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(rtracklayer)

plot_or = function(dat, title, xlab="Odds ratio", ylab="Annotations") {
  p1 <- ggplot(dat, aes(y=annotation)) + 
    geom_point(aes(x=logOR), colour="#800000") + #plot OR
    geom_segment(aes(x=lower,xend=upper,yend=annotation), alpha=0.8)
  p1 <- p1 + geom_vline(xintercept=0) + theme_bw() + labs(title=title, x=xlab, y=ylab)
  p1
}
```

## Reproducing Fig3b in the ASoC paper
![](assets/reproduced_Fig3B.png)

<br><br><br>
![](assets/ASoC_Fig3B.jpeg)


## Schizophrenian PGC3

PGC3 data from <https://www.medrxiv.org/content/10.1101/2020.09.12.20192922v1>. 

There are 69,369 cases and 236,642 controls in total. 7585077 SNPs with MAF>1% were included.

After dropping X,Y Chromosome and removing duplicate, SNPs 7513220 SNPs remain.


## FDR correction

The original FDR is effectively using Bonferroni correction. 
```{r correction, echo=FALSE}
bedd <- import.bed("data/NpCm_glut_6v0hr.bed")
pval <- sapply(strsplit(bedd$name, "\\^"), function(x) {as.numeric(x[2])})
qval <- sapply(strsplit(bedd$name, "\\^"), function(x) {as.numeric(x[3])})
plot(qval/pval, xlab=paste0("Peaks in NpCm_glut_6v0hr.bed sorted by pvalues"), ylab="FDR/P")
```

## Summary for different peak sets

* Number of peaks in FWER<0.25 differential peaks
```{r, echo=FALSE}
peak.counts.bonf25 <- read.table("data/peak_counts/FDR0.25.txt", header=T)
knitr::kable(peak.counts.bonf25[,c("bed","peak")])
```
<br><br>

* Number of peaks called for each cell type at each timepoint
```{r table_12peaks, echo=FALSE}
peak.counts.12peaks <- read.table("data/peak_counts/12peaks.txt", header=T)
knitr::kable(peak.counts.12peaks)
```
<br><br>
* The number of peaks in the last paper
```{r, echo=FALSE}
peak.counts.ASoC <- read.table("data/peak_counts/peak_counts_old_ASoC.txt", header=T)
knitr::kable(peak.counts.ASoC)
```
<br><br>
* The proportion of 0hr single cell peaks that overlaps with bulk peaks:
```{r, echo=FALSE}
entailment <- read.table("data/sc_vs_bulk.txt", header=T, row.names = 1)
rownames(entailment) <- c("GABA_0hr", "NmCp_glut_0hr", "NPC_0hr", "NpCm_glut_0hr")
colnames(entailment) <- c("CN", "DN","GA","ips","NPC")
knitr::kable(entailment)
```
<br><br>
* The number of peaks selected with FDR<0.05:
```{r table_DAP_bothside, echo=FALSE}
peak.counts.FDR5 <- read.table("data/peak_counts/FDR0.05.txt", header=T)
knitr::kable(peak.counts.FDR5[c('bed','peak')])
```
<br><br>
* The number of peaks selected with FDR<0.05 & log(fold change)>0:
```{r table_DAP_positive, echo=FALSE}
peak.counts.FDR5.pos <- read.table("data/peak_counts/FDR0.05_positive.txt", header=T)
knitr::kable(peak.counts.FDR5.pos[c('bed','peak')])
```
<br><br>
* Peaks in resting condition with at least 1bp overlap with DAPs (6v0 or 1v0) were removed. Actually a few hundred peaks have been removed.
```{r table_static, echo=FALSE}
peak.counts.static <- read.table("data/peak_counts/static_bothside.txt", header=T)
knitr::kable(peak.counts.static)
```
<br><br>
* Peaks called directly through MACS2 by merging bam files within the same cell type and the same time point. Called pseudo bulk peaks here.
```{r table_pseudobulk, echo=FALSE}
peak.counts.pseudo <- read.table("data/peak_counts/peak_counts_pseudobulk.txt", header=T)
knitr::kable(peak.counts.pseudo)
```
<br><br>
* Remove differential peaks from the pseudo bulk peaks
```{r table_static_pseudobulk, echo=FALSE}
peak.counts.static.pseudo <- read.table("data/peak_counts/peak_counts_static_pseudobulk.txt", header=T)
knitr::kable(peak.counts.static.pseudo)
```